app/models/detection.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# A Detection is a joining record between a Term and a Detector, created when a Detector activates based on some aspect
+# of the Term. This is the record that TACOS found something about this Term.
+#
+# There is a uniqueness constraint on the combination of term_id, detector_id, and detector_version.
+#
+# New records can be created by passing a Term and a Detector object. The model will look up the current detector
+# version, and include that in the record. If this would result in a duplicate record, then the creation process is
+# aborted with no duplicate created.
+#
+# == Schema Information
+#
+# Table name: detections
+#
+#  id               :integer          not null, primary key
+#  term_id          :integer          not null
+#  detector_id      :integer          not null
+#  detector_version :string
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#
+class Detection < ApplicationRecord
+  belongs_to :term
+  belongs_to :detector
+
+  # We use the before_create hook to prevent needing to override the initialize method, which Rails frowns upon.
+  before_create :set_defaults, :check_for_current
+
+  # These scopes allow for easy filtering of Detection records by a single parameter.
+  scope :current, -> { where(detector_version: ENV.fetch('DETECTOR_VERSION', 'unset')) }
+  scope :for_detector, ->(detector) { where(detector_id: detector.id) }
+  scope :for_term, ->(term) { where(term_id: term.id) }
+
+  private
+
+  # This leverages three scopes to confirm, before creating a new Detection, whether that combination of Term, Detector,
+  # and Version has already been registered. If one is found, it aborts the creation.
+  def check_for_current
+    return unless Detection.for_term(term).for_detector(detector).current.exists?
+
```

Comment:
I'm trying to understand a use case where this logic is preferred over using a `Detection.find_or_create_by` to always return a `Detection` (either an existing one ore creating and returning the new one).

Reply:
That's an interesting thought, and it would allow us to avoid this method. The biggest drawback I can come up with is that `find_or_create_by` actually returns a record - and these methods don't actually do anything with that - they don't return anything, and take no actions after creating the records.

Is that enough of a reason not to use `find_or_create_by`? I'm not sure. If you'd rather go that route, I'm happy to give it a shot. We could probably ditch all three scopes and this private method (the other one would still be needed, though).

Reply:
The more I think about this, the more I think it makes sense to use `find_or_create_by` in this context. I think there might still be a need for the scopes, but that's just a suspicion rather than anything we've discussed clearly.

I'll take this back out of review and push an updated commit soon.